ember-toucan-core/src/components/form/wrapper.hbs

```diff
@@ -0,0 +1,15 @@
+{{#if (has-block)}}
+  {{yield
+    (hash
+      Label=(component this.Label)
+      Hint=(component this.Hint)
+      Control=(component this.Control)
+      Error=(component this.Error)
+    )
+  }}
+{{else}}
+  {{yield (component this.Label) to="label"}}
+  {{yield (component this.Hint) to="hint"}}
+  {{yield (component this.Control) to="control"}}
+  {{yield (component this.Error) to="error"}}
```

Comment:
One of the things @nicolechung and I talked about, but I'm not personally _quite_ sure provides a ton of value is yielding back named blocks here versus only doing the contextual components.

So if you do:

```hbs
<Field>
  <:label>Label<:/label>
</Field>
```

You'll still get the hint + error divs rendered like https://github.com/CrowdStrike/ember-toucan-core/issues/10#issuecomment-1422872713 mentions, so you'll get extra stuff in the DOM that you don't (potentially) care about.  And in the case of the Error, you'll have a lingering error icon.

I _feel_ like the contextual component path is a better API for consumers, as folks can add only what they need like:


```hbs
// Taken from https://github.com/CrowdStrike/ember-toucan-core/compare/10-field-prototype-w-nicole#diff-6b74531eede9b2af7db5be79e6cc466ce08e9f1c7c604348a819b3a0825f8d76
<template>
  <Field as |field|>
    <field.Label>{{@label}}</field.Label>

    {{#if @helperText}}
      <field.Hint>{{@helperText}}</field.Hint>
    {{/if}}

    <field.Control>
      <input class="border-critical bg-blue" ...attributes />
    </field.Control>

    {{#if @error}}
      <field.Error>{{@error}}</field.Error>
    {{/if}}
  </Field>
</template>
```

and then usage of above:


```hbs
<InputField @label="Input" />
<InputField @label="Input" @helperText="help!" />
<InputField @label="Input" @helperText="help!" @error="error" />
```

---

I also don't have strong opinions, I think it's just more my brain trying to figure out a use case for the named blocks if they _can't_ conditionally render blocks they don't care about.  Any thoughts around this or guidance?

Reply:
For conditionally rendering named blocks, wouldn't something like this work?

```hbs
{{#if (has-block "label")}}
  {{yield to="label"}}
{{/if}}
```

But this is not necessarily better than yielded components...

The way I would think about this is who is in charge of the overall markup and layout? 

If we have strong opinions/guidelines/rules for where to put which element, or having markup that wraps everything, that shouldn't be user-controllable, then named blocks are the better fit. They are more restrictive, or let you be in control, which I think is good for design systems, as that's what they do: they impose restrictions on the what and how.

On the other side, if you want to be less opinionated, and give the user freedom to compose things in arbitrary ways, then yielded components are good. headless-form is a good example of this latter case, it's by its own definition un-opinionated. 

So not sure how the field components fits in here. Some questions, which also might help to understand how opinionated this component needs to be:
* what's the deal with the "hint"? Is it really the case that the user can put them either before or after the control, or doesn't the design system dictate where the should be positioned, like ? (unfortunately, I wasn't able to take part in the latest sessions. Wasn't there cases where the hint was above the control, aligned to the right?)
* don't we need to wrap the whole field in some markup, like tailwind flex classes or something, to position things properly according to our Figma designs? Or is this not a concern of this component?

Reply:
Sounds great! Thanks for the thorough explanation, especially related to the "what is the concern of this" question!

> I'm curious if you think there's too much overlap between this and what ember-headless-form provides 

Yeah, I am seeing that a bit at least. But I guess we cannot prevent this, when Field should also be usable standalone, without a full-fledged form. I also think this is not terrible.